### ISSUE TITLE
Web UI inaccessible

### ISSUE THREAD

**Issue by mikolajpp** [urbit/urbit#3015]
**Describe the bug**
After a day of running the http UI has become inaccessible. 
Sporadic `http`'s are no longer visible in the dojo.

**System (please supply the following information, if relevant):**
 - OS: linux64
 - Urbit 0.10.5
 - `0vu.niou0.tga8g.m8jo3.pbpr0.f3ih0.vkkhu.a0i7o.dhntn.82tm9.81dn2`

Additional my home is out of sync with base and is of hash 
`%home-hash 0v4.0eh6c.nvtqb.joh0t.tb89j.dcaug.5vtlk.1h51j.sh1b4.9t988.9vqqk`

**Comment by mikolajpp** [comment]
The web UI unlocked itself with no apparent signs.
Base and home are still out of sync. All I can say about this for now that 
during the hang the HTTP port was not even open, as per `netstat -tup` output. 

Now that it works I can see in the list.

**Comment by mikolajpp** [comment]
After waking computer from sleep web UI no longer works. Here is the output of `netstat -tup | grep urbit`

```
tcp        1      0 127.0.0.1:8080          127.0.0.1:36470         CLOSE_WAIT  25771/urbit          off (0.00/0/0)
tcp        1      0 127.0.0.1:8080          127.0.0.1:36252         CLOSE_WAIT  25771/urbit          off (0.00/0/0)
tcp        1      0 127.0.0.1:8080          127.0.0.1:36412         CLOSE_WAIT  25771/urbit          off (0.00/0/0)
tcp        1      0 127.0.0.1:8080          127.0.0.1:36468         CLOSE_WAIT  25771/urbit          off (0.00/0/0)
tcp        1      0 127.0.0.1:8080          127.0.0.1:36250         CLOSE_WAIT  25771/urbit          off (0.00/0/0)
tcp        1      0 127.0.0.1:8080          127.0.0.1:36362         CLOSE_WAIT  25771/urbit          off (0.00/0/0)
tcp        1      0 127.0.0.1:8080          127.0.0.1:36416         CLOSE_WAIT  25771/urbit          off (0.00/0/0)
tcp        1      0 127.0.0.1:8080          127.0.0.1:36254         CLOSE_WAIT  25771/urbit          off (0.00/0/0)
tcp        1      0 127.0.0.1:8080          127.0.0.1:36466         CLOSE_WAIT  25771/urbit          off (0.00/0/0)
tcp        1      0 127.0.0.1:8080          127.0.0.1:36270         CLOSE_WAIT  25771/urbit          off (0.00/0/0)
tcp        1      0 127.0.0.1:8080          127.0.0.1:36382         CLOSE_WAIT  25771/urbit          off (0.00/0/0)
tcp        1      0 127.0.0.1:8080          127.0.0.1:36462         CLOSE_WAIT  25771/urbit          off (0.00/0/0)
```

So you see a lot of connections which are supposed to be closed, but urbit keeps them hanging.  These must be the connections resulting from frantic trials to open the web UI in the browser, which never succeded. 

So far I have no reproducible solution for this. It seems to happen after prolonged time of not accessing the Web UI and also after sleep/wake cycles.

**Comment by mikolajpp** [comment]
Furthermore, after restartin the urbit process, I can see the following:
```
[%canceling-nonexistent-channel ~[//http-server/0vi.jt54i/74/6]]
[%canceling-nonexistent-channel ~[//http-server/0vi.jt54i/47/1]]
[%canceling-nonexistent-channel ~[//http-server/0vi.jt54i/73/1]]
[%canceling-nonexistent-channel ~[//http-server/0vi.jt54i/5/1]]
[%canceling-nonexistent-channel ~[//http-server/0vi.jt54i/59/4]]
[%canceling-nonexistent-channel ~[//http-server/0vi.jt54i/61/2]]
[%canceling-nonexistent-channel ~[//http-server/0vi.jt54i/4/8]]
[%canceling-nonexistent-channel ~[//http-server/0vi.jt54i/3/3]]
[%canceling-nonexistent-channel ~[//http-server/0vi.jt54i/40/1]]
[%canceling-nonexistent-channel ~[//http-server/0vi.jt54i/75/4]]
[%canceling-nonexistent-channel ~[//http-server/0vi.jt54i/11/2]]
[%canceling-nonexistent-channel ~[//http-server/0vi.jt54i/71/6]]
```

In total 12 connections closed, exact same number as the output from netstat above; indeed these connections are now gone, as netstat shows, but the problem is there aren't any connections opened by urbit. 

In short, there is some strange bug which periodically messes with the open sockets. 

The vere binary I am running was compiled on site and is from the latest official revision, 0.10.5

**Comment by mikolajpp** [comment]
Okay, I have found the culprit. This is a new machine and for some reason 
the setup wizard set the hostname, but did not put the entry for the hostname into `/etc/hosts`
which must have resulted in name resolution errors on urbit side. 

Once I have added 127.0.0.1 entry to the /etc/hosts web UI is present just fine. 

Although this is clearly a host misconfiguration issue, some of the time web UI worked, some of the time it did not work with no apparent error messages. It would be better to just fail in a deterministic fashion.
